blob/store.go

```diff
@@ -62,6 +70,28 @@ type StoreCloser interface {
 	Closer
 }
 
+// KVCore is the common interface shared by implementations of a key-value
+// namespace. Users will generally not use this interface directly; it is
+// included by reference in [KV] and [CAS].
+type KVCore interface {
+	// Get fetches the contents of a blob from the store. If the key is not
+	// found in the store, Get must report an ErrKeyNotFound error.
+	Get(ctx context.Context, key string) ([]byte, error)
+
+	// Delete atomically removes a blob from the store. If the key is not found
+	// in the store, Delete must report an ErrKeyNotFound error.
+	Delete(ctx context.Context, key string) error
+
+	// List calls f with each key in the store in lexicographic order, beginning
+	// with the first key greater than or equal to start.  If f reports an error
+	// listing stops and List returns.  If f reported an ErrStopListing error,
+	// List returns nil; otherwise List returns the error reported by f.
+	List(ctx context.Context, start string, f func(string) error) error
```

Comment:
Not for this change, but further "for later" thought: this wants to be an iter.Seq, I think!

Reply:
Maybe. I thought about it when iterator support got released—but then there is no good way for `f` to do something complex with a key (e.g., load its data, delete it, do some other lookups with it, etc.) and report a fatal error back to `List`.

Reply:
(will defer this question for later, but added #31)